pkg/validator/validators.go

```diff
@@ -76,13 +86,25 @@ func trafficTargetValidator(req *admissionv1.AdmissionRequest) (*admissionv1.Adm
 }
 
 // ingressBackendValidator validates the IngressBackend custom resource
-func ingressBackendValidator(req *admissionv1.AdmissionRequest) (*admissionv1.AdmissionResponse, error) {
+func (kc *policyValidator) ingressBackendValidator(req *admissionv1.AdmissionRequest) (*admissionv1.AdmissionResponse, error) {
 	ingressBackend := &policyv1alpha1.IngressBackend{}
 	if err := json.NewDecoder(bytes.NewBuffer(req.Object.Raw)).Decode(ingressBackend); err != nil {
 		return nil, err
 	}
 
+	type backendCacheKey struct {
+		name string
+		port int
+	}
+
+	backends := make(map[backendCacheKey]struct{}, len(ingressBackend.Spec.Backends))
```

Comment:
Use `mapset` here. If `mapset.Add()` returns true, it implies the item was added and does not contain a duplicate.